### DISCUSSION TITLE
How to update just a field in the database

### DISCUSSION THREAD

**Discussion by rhuanbarros** [supabase-community/supabase-csharp#49] [Q&A] [answered]
I don't know if I'm using the API correctly, but I need to update just a field in a row in the database. I tryed to use the method
`await client.From<TModel>().Update(item);`
But the Update method receveis a object of type BaseModel.
`Task<ModeledResponse<T>> Update(T model, QueryOptions? options = null, CancellationToken cancellationToken = default);`

So this way I need to declare a whole new class with just the field that I need to update in the database or retrive all the data in another call and after this, update the row sending all the data of the row?

Sorry, if I get it all wrong, but I'm really confusied. I'm stuck with it.

In the context of SQL querys we can update just the field that we need, but I can't figure it out how to do it with supbase-csharp api.

**Comment by acupofjose** [comment, marked as answer]
Hey @rhuanbarros, at the moment I’m writing a PR that addresses this exact issue. I plan on having a new version with this functionality to work out in the next day or two. 

But, yes, at the moment you would have to have get the model, update the value locally, and then update it. It’s a rather un-ideal setup. New changes will make this much easier!

you can follow the progress here: supabase-community/postgrest-csharp#60

**Reply by rhuanbarros** [reply]
Awesome, can't wait to see the new version with this functionality! Keep up the great work!

**Reply by acupofjose** [reply]
Available in 0.7.0!

**Reply by rhuanbarros** [reply]
Hi,
I'm testing the new version and find that the code to update a field didn't work with a field of type bool. It just works when I change the type of the field to string.

```
// class model definition file
[Column("SoftDelete")]
    public bool? SoftDelete { get; set; }  // type bool don't work. Throw Unhandled exception rendering component: Key should reference a Model Property.
    // public string? SoftDelete { get; set; }  // type string works fine
```


```
// call to database api file
await client.Postgrest
			.Table<Todo>()
            .Set(x => new KeyValuePair<object,object>(x.SoftDelete, true))
            // .Set(x => new KeyValuePair<object,object>(x.SoftDelete, "true"))
            .Where(x => x.Id == item.Id)
            .Update();

```

I uploaded a test project to this link to see the whole code working:
https://github.com/rhuanbarros/BlazorWebAssemblySupabaseTemplate/blob/main/Dtos/Todo.cs
https://github.com/rhuanbarros/BlazorWebAssemblySupabaseTemplate/blob/main/Services/DatabaseService.cs

**Reply by acupofjose** [reply]
Fix is available in 3.1.1 - I've also shortened the accessor for `Set` so you can do `.Set(x => x.SoftDelete, true)`

Additional tests were added to PostgrestTests/Linq.cs if you're interested! Thanks for testing, always appreciate how detailed your error reproductions are. Makes my life easier!

**Reply by rhuanbarros** [reply]
The code is working now. Thanks for providing the update in the speed of light! 
And thanks for creating this library, it's helping me a lot.

I will keep updating the template project with some of the new features in my spare time.
